pkg/util/kubeconfig/kubeconfig.go

```diff
@@ -57,36 +55,32 @@ func WriteKubeConfig(currentNamespaceConfig *rest.Config, secretName, secretName
 		key := clientConfig.KeyData
 
 		kubeConfigSecret := &corev1.Secret{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "Secret",
-				APIVersion: "v1",
-			},
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      secretName,
 				Namespace: secretNamespace,
 			},
-			Type: corev1.SecretTypeOpaque,
-			Data: map[string][]byte{
-				KubeconfigSecretKey:     out,
-				CADataSecretKey:         caData,
-				CertificateSecretKey:    cert,
-				CertificateKeySecretKey: key,
-			},
 		}
 
 		// set owner reference
 		if translate.Owner != nil && translate.Owner.GetNamespace() == kubeConfigSecret.Namespace {
 			kubeConfigSecret.OwnerReferences = translate.GetOwnerReference(nil)
 		}
```

Comment:
Owner references need to be set inside the CreateOrPatch mutator function.

Reply:
@matskiv good catch! Fixed!